js/src/util/index.js

```diff
@@ -107,7 +107,11 @@ const emulateTransitionEnd = (element, duration) => {
   const durationPadding = 5
   const emulatedDuration = duration + durationPadding
 
-  function listener() {
+  function listener({ target }) {
+    if (target !== element) {
+      return
+    }
```

Comment:
The same thing is done in the `executeAfterTransition` function as well :thinking:

Needed on both places?

Reply:
I'm sorry but I guess it does. Both `emulateTransitionEnd` and `executeAfterTransition` suffer from the problem that `transitionend` events do bubble. So without this some nested event can interupt the intended behaviour.

Both functions could be merged together but that would have required some change to the signature of `emulateTransitionEnd` (or to be renamed because it would do more than just emulating events). I wasn't sure if this could be some breaking change for 3rd party code that relies on our utilities, so I only added this fix.

IMHO `emulateTransitionEnd` could be replaced at some point anyway because the browsers we target, all do support `transitionend` events natively from what I can tell, so there is no need to use `setTimeout`.

The problem is by removing `setTimeout` around 80 tests fail, because we're not using css in our unit tests and the tests rely on that functionality.